src/SwiftFormatEditProvider.ts

```diff
@@ -132,10 +132,24 @@ export class SwiftFormatEditProvider
     ch: string,
     formatting: vscode.FormattingOptions,
   ) {
-    // Don't format if user has inserted an empty line
-    if (document.lineAt(position.line).text.trim() === "") {
-      return [];
+    // If a new line was entered, format the previous line
+    if (ch === "\n" && position.line > 0) {
+      const previousLine = position.line - 1;
+
+      // If the previous line is empty, formatting is not required
+      if (document.lineAt(previousLine).text.trim() === "") {
+        return [];
+      }
+
+      // Create a range for the previous line and format it
+      const previousLineText = document.lineAt(previousLine).text;
+      const range = new vscode.Range(new vscode.Position(previousLine, 0), new vscode.Position(previousLine, previousLineText.length));
+      return format({
+        document,
+        parameters: ["--fragment", "true"],
+        range,
+        formatting
+      });
     }
-    return format({ document, formatting });
```

Comment:
looks good but i would do something like:
```
if (ch === "\n") {
  if (position.line > 0) {
    // .... your code here
  } else {
    return [];
  }
} else {
    return format({ document, formatting });
}
```
This just ensures that the function will return something in all paths.
Currently the character can only be `\n` some some of this code will never be triggered, but i'd rather be safe than sorry considering the code can change in the future too.

Reply:
Changed.